docs/HTTP-batchsource.md

```diff
@@ -0,0 +1,206 @@
+
+**Non-HTTP Error Handling:** Strategy used to handle errors during transformation of a text entry to record.
+Possible values are:<br>
+Stop on error - Fails pipeline due to erroneous record.<br>
+Send to error - Sends erroneous record's text to error port and continues.<br>
+Skip on error - Ignores erroneous records.
+
+**Retry Policy:** Policy used to calculate delay between retries.
+
+**Linear Retry Interval:** Interval between retries. Is only used if retry policy is "linear".
+
+**Max Retry Duration:** Maximum time in seconds retries can take.
+
+**Connect Timeout:** Maximum time in seconds connection initialization is allowed to take.
+
+**Read Timeout:** Maximum time in seconds fetching data from the server is allowed to take.
+
+### Pagination
+
+**Pagination Type:** Strategy used to determine how to get next page.
+
+**Wait time between pages:** Interval between loading pages in milliseconds.
+<br><br>
+
+##### Pagination type: None
+Only single page is loaded.
+<br><br>
+
+##### Pagination type: Link in response header
+In response there is a "Link" header, which contains an url marked as "next". Example:<br>
+`<http://example.cdap.io/admin/api/pages?page=1&q.language.id=1>; rel="first",
+ <http://example.cdap.io/admin/api/pages?page=2&q.language.id=1>; rel="next",
+ <http://example.cdap.io/admin/api/pages?page=2&q.language.id=1>; rel="last"`
+<br><br>
+
+##### Pagination type: Link in response body
+Every page contains a next page url. This pagination type is only supported for JSON and XML formats.
+Pagination happens until no next page field is present or until page contains no elements.
+
+**Next Page JSON/XML Field Path:** A JSON path or an XPath to a field which contains next page url.
+<br><br>
+
+##### Pagination type: Token in response body
+Every page contains a token, which is appended as an url parameter to obtain next page.
+This type of pagination is only supported for JSON and XML formats. Pagination happens until no next page
+token is present on the page or until page contains no elements.
+
+**Next Page Token Path:** A JSON path or an XPath to a field which contains next page token.
+
+**Next Page Url Parameter:** A parameter which is appended to url in order to specify next page token.
+<br><br>
+
+##### Pagination type: Increment an index
+Pagination by incrementing a {pagination.index} placeholder value in url. For this pagination type url is required
+to contain above placeholder.
+
+**Start Index:** Start value of {pagination.index} placeholder
+
+**Max Index:** Maximum value of {pagination.index} placeholder. If empty, pagination will happen until the page with
+no elements.
+
+**Index Increment:** A value which the {pagination.index} placeholder is incremented by. Increment can be negative.
+<br><br>
+
+##### Pagination type: Custom
+Pagination using user provided code. The code decides how to retrieve a next page url based on previous page contents
+and headers and when to finish pagination.
```

Comment:
Should include an example to make it more clear.

Reply:
I'll include the example for this once I implement this. Currently don't have enough info.